packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -571,11 +633,27 @@ export function acquireResource(resource: Resource): Instance {
   }
 }
 
-export function releaseResource(resource: Resource) {
+export function releaseResource(resource: Resource): void {
   switch (resource.type) {
+    case 'head': {
+      return releaseHeadResource(resource);
+    }
     case 'style': {
       resource.count--;
+      return;
+    }
+  }
+}
+
+function releaseHeadResource(resource: HeadResource): void {
+  if (--resource.count === 0) {
+    // the instance will have existed since we acquired it
+    const instance: Instance = (resource.instance: any);
+    const parent = instance.parentNode;
+    if (parent) {
+      parent.removeChild(instance);
```

Comment:
Is there a test that ensures that if you remove this and then reacquire the same resource later in the mutation phase, that it gets readded?